src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.cs

```diff
@@ -134,12 +134,31 @@ protected void SkipToAndParseCode(Func<HtmlSymbol, bool> condition)
                 {
                     if (last != null)
                     {
-                        Accept(last);
+                        if (!startOfLine || last.Type != HtmlSymbolType.WhiteSpace)
```

Comment:
If not for this condition, the whitespace preceding the razor comment will show up as part of the next line.